### ISSUE TITLE
Question about computation time

### ISSUE THREAD

**Issue by dongyuanwu** [ZJUFanLab/SpaTalk#14]
Hi,

Thanks for this useful package!

I have a question about the computation time. In the supplementary document of the paper, you provided two tables of the computation times, and they looked pretty good. I am trying to analyze the same 10X Visium data in your paper (i.e., P2.rep2). However, `dec_celltype()` has taken a pretty long time, and still no response after printing "Performing Non-negative regression for each spot". I would just like to check the number of cores and memory you used for getting efficient results in supplementary tables. I didn't find this information in the paper, not sure if I missed something.

Looking forward to your reply and thank you in advance!

**Comment by multitalk** [comment]
Thanks for your question. In fact, we compared the deconvolution time with other methods in our paper, namely the "Performing Non-negative regression for each spot" . However, there is another step called generating single-cell ST atlas for spot-based ST data which might take a long time for a large dataset. 
We used a server with 64 cores and 256G memory for benchmarking. Here is the normal process for  `dec_celltype()`:

```
> obj <- dec_celltype(object = obj, sc_data = as.matrix(starmap_data),  sc_celltype = starmap_meta$celltype)
Performing Non-negative regression for each spot 
Generating single-cell data for each spot 
***Done*** 
> 
```
